pallets/tendermint-light-client/src/lib.rs

```diff
@@ -0,0 +1,337 @@
+#![cfg_attr(not(feature = "std"), no_std)]
+
+//! This pallet is an on-chain light-client for tendermint (Cosmos) based chains
+//! It verifies headers submitted to it via on-chain transactions, performed by a so-called relayer
+//! It is a part of the Aleph Zero <-> Terra bridge
+pub use pallet::*;
+
+pub mod types;
+mod utils;
+
+use frame_support::traits::StorageVersion;
+
+/// The current storage version.
+const STORAGE_VERSION: StorageVersion = StorageVersion::new(0);
+
+#[frame_support::pallet]
+pub mod pallet {
+    use super::*;
+    use crate::types::{
+        LightBlockStorage, LightClientOptionsStorage, TendermintBlockHash, TendermintHashStorage,
+    };
+    use frame_support::{
+        ensure, fail, log,
+        pallet_prelude::{
+            DispatchClass, DispatchResult, IsType, OptionQuery, StorageMap, StorageValue,
+            ValueQuery,
+        },
+        traits::{Get, UnixTime},
+        Identity,
+    };
+    use frame_system::{ensure_root, ensure_signed, pallet_prelude::OriginFor};
+    use tendermint::Time;
+    use tendermint_light_client_verifier::{
+        options::Options, types::LightBlock, ProdVerifier, Verdict, Verifier,
+    };
+
+    #[pallet::config]
+    pub trait Config: frame_system::Config {
+        /// ubiquitous event type
+        type Event: From<Event<Self>> + IsType<<Self as frame_system::Config>::Event>;
+
+        /// Maximal number of finalized headers to keep in the storage, last-in first-out
+        #[pallet::constant]
+        type HeadersToKeep: Get<u32>;
+
+        /// Maximal number of block validators in Tendermint
+        #[pallet::constant]
+        type MaxVotesCount: Get<u32>;
+
+        /// time provider type, used to gauge whether blocks are within the trusting period
+        type TimeProvider: UnixTime;
+    }
+
+    #[pallet::pallet]
+    #[pallet::storage_version(STORAGE_VERSION)]
+    pub struct Pallet<T>(_);
```

Comment:
`T` is not used, can we remove it?

Reply:
it is used, just obscured by the attribute macros